forrustts-tables-trees/src/trees.rs

```diff
@@ -96,15 +96,20 @@ impl NodeIterator for PreorderNodeIterator<'_> {
         self.current_node_ = self.node_stack.pop();
         match self.current_node_ {
             Some(u) => {
-                let mut c = self.tree.left_child(u).unwrap();
+                let mut c = self.tree.right_child(u).unwrap();
```

Comment:
We need to add tests that we get the exact order out that we expect, correct number and ID of roots in each tree, etc..